homeassistant/components/rflink.py

```diff
@@ -166,15 +167,17 @@ def event_callback(event):
         if entities:
             # Propagate event to every entity matching the device id
             for entity in entities:
-                _LOGGER.debug('passing event to %s', entities)
-                entity.handle_event(event)
+                _LOGGER.debug('passing event to %s', entity)
+                async_dispatcher_send(hass, SIGNAL_HANDLE_EVENT, entity, event)
         else:
-            _LOGGER.debug('device_id not known, adding new device')
-
             # If device is not yet known, register with platform (if loaded)
             if event_type in hass.data[DATA_DEVICE_REGISTER]:
+                _LOGGER.debug('device_id not known, adding new device')
+                # TODO: Is there a race condition here if we get another event before async_run_job is done?
```

Comment:
One solution is to add the device id to the registry but not add any entity ids to the list that the device id key holds in the registry. Do this here to prevent us from adding another entity before the entity has need added.